### PR TITLE
PP 30 Add a new class, render teacher's classes

### PR DESCRIPTION
**Pull request by vrudnitskaya**
### Description
Implemented the addition of a new class to the teacher's dashboard. All teacher classes are now also displayed.
### Screenshots
filled-out-form

After adding a class, the user is redirected to a page with all classes.
added-class

There is validation of the form before it is sent to the server.
form-errors

If the form is submitted to the server and a validation error occurs, the error is also displayed.
form-errors2

### CODE REVIEW

**Comment by Tanyaa-a** [comment]
Great work Valya! The add class form works great from my side!

File: src/components/common/FormInput.jsx
Comment (Tanyaa-a) [inline src/components/common/FormInput.jsx:1]: The add new class works good from my side! Awesome work Valya!

**Review by Tanyaa-a — APPROVED** [review]

**Review by AnnaPestova1 — CHANGES_REQUESTED** [review]
Hi Valya! Great job. Just a small issue, could we handle error message after we added category? 
image
here the full form after I added the info. I still have an error message in the category input 
image

and why we cannot add babies? 
image

File: src/components/pages/dashboard-pages/AddClass.jsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    goal: '',
+    experience: '',
+    other: '',
+    availableTime: { date: '', startTime: '' },
+  });
+  const [formErrors, setFormErrors] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
```
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/AddClass.jsx:31]: Great function!

**Comment by vrudnitskaya** [comment]
> 
> 
> Can you please take a look, do we need an error message if user does not have classes yet? (maybe we need something like: "add your first class" message instead? image

fixed it
no-classes-teacher

**Review by AnnaPestova1 — APPROVED** [review]
Great work Valya! 

Please solve conflicts and check functionality before merging. 

I think we have a redundancy here, just take a look do we need both - the left and right sides to add the first class, or we can show the left part if classes exist?
image

**Comment by vrudnitskaya** [comment]
> Great work Valya!
> 
> Please solve conflicts and check functionality before merging.
> 
> I think we have a redundancy here, just take a look do we need both - the left and right sides to add the first class, or we can show the left part if classes exist? image

no-classes

**Pull request merged**